src/ctl.c

```diff
@@ -3457,3 +3467,34 @@ experimental_arenas_i_pactivep_ctl(tsd_t *tsd, const size_t *mib,
 	malloc_mutex_unlock(tsd_tsdn(tsd), &ctl_mtx);
 	return ret;
 }
+
+static int
+experimental_prof_recent_alloc_max_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+
+	if (!(config_prof && opt_prof)) {
+		ret = ENOENT;
+		goto label_return;
+	}
+
+	ssize_t old_max;
+
+	if (newp != NULL && newlen == sizeof(ssize_t)) {
```

Comment:
This isn’t the right way to handle write. If the newlen doesn’t match, an error should be reported. See https://github.com/jemalloc/jemalloc/blob/dev/src/ctl.c#L1465

Use WRITE instead.

Reply:
Makes sense. Put `WRITE()` inside the `newp != NULL` branch; otherwise it's hard to determine whether the caller's intention is read/write or read-only.

Also added relevant unit tests.